zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
```

Comment:
@ssanderson I recommended that David implement one 'Corrlation' factor with an argument to specify which type of correlation is being computed. Pearson is the default in nearly every analysis so it makes sense to have that be the default one. In terms of issues with normality, yes but that's something that a user is just going to have to learn through the lectures imo.